src/DynamoUtilities/ThreadSafeList.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace DynamoUtilities
+{
+    /// <summary>
+    /// Thread safe List<T>
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class ThreadSafeList<T> : IList<T>
```

Comment:
I find it strange that we have to innovate our very own data structures and cannot use something off-the-shelf here?

Reply:
There are some thread safe collections of sorts out of the box..,but none that implement IList interface

I chose to create a thread safe IList so that I could replace the inner IList in SmartObservableCollection.ObservableCollection.Collection base class.